### PR TITLE
LCAM-1285: Automate multi-module release workflow for crime commons

### PR DESCRIPTION
**Pull request by skasthuri36326**
## What

[LCAM-1285](https://dsdmoj.atlassian.net/browse/LCAM-1285)

Describe what you did and why.
- Updated build with semver gradle plugin for multi-module release
- Updated circleci workflows to segregate the release steps crime commons rest client and crime-common-classes 
- updated nexusPublishing to publish snapshot and release versions

[LCAM-1285]: https://dsdmoj.atlassian.net/browse/LCAM-1285?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -25,7 +25,7 @@ executors:
 # COMMANDS
 # ------------------
 commands:
-  publish_artefact:
+  publish_artefact_rest_client:
```
Comment (Mhowell494) [inline .circleci/config.yml:28]: Is there any chance we can combine the publish commands? We're already passing the release type, so why not pass the artefact/module name and consolidate them.

File: crime-commons-classes/build.gradle [hunk 2]
```diff
@@ -1,12 +1,15 @@
-
 plugins {
+    id "com.javiersc.semver.project" version "0.5.0"
     id 'java'
```
Comment (Mhowell494) [inline crime-commons-classes/build.gradle:3]: Can we clean this up a little?
This plugin can be removed as it's defined in the parent `build.gradle`.
The `java` closure can also be removed.

**Review by Mhowell494 — CHANGES_REQUESTED** [review]
Just a few minor points to consider.

**Comment by Mhowell494** [comment]
Can we make sure we squash the commits when we merge please? :)

File: .circleci/config.yml (hunk 1 above)
Comment (skasthuri36326) [reply .circleci/config.yml:28]: I haven't tried passing parameter from circleci jobs to commands. Will test and let you know.

**Review by Mhowell494 — APPROVED** [review]

**Pull request merged**